src/protocol/cflie_nrf24l01.c

```diff
@@ -390,20 +668,40 @@ static int cflie_init()
     return 50000;
 }
 
-
-#ifdef CFLIE_TELEMETRY
-static void update_telemetry() {
-  static u8 frameloss = 0;
-
-  frameloss += NRF24L01_ReadReg(NRF24L01_08_OBSERVE_TX) >> 4;
-  NRF24L01_WriteReg(NRF24L01_05_RF_CH, rf_channel);   // reset packet loss counter
-
-  Telemetry.p.dsm.flog.frameloss = frameloss;
-//  Telemetry.p.dsm.flog.volt[0] = read battery voltage from ack payload
-  TELEMETRY_SetUpdated(TELEM_DSM_FLOG_FRAMELOSS);
+static void update_telemetry()
+{
+    static u8 frameloss = 0;
+
+    // Read and reset count of dropped packets
+    frameloss += NRF24L01_ReadReg(NRF24L01_08_OBSERVE_TX) >> 4;
+    NRF24L01_WriteReg(NRF24L01_05_RF_CH, rf_channel); // reset packet loss counter
+    Telemetry.value[TELEM_DSM_FLOG_FRAMELOSS] = frameloss;
+    TELEMETRY_SetUpdated(TELEM_DSM_FLOG_FRAMELOSS);
+
+    if (packet_ack() == PKT_ACKED) {
+        // See if the ACK packet is a cflie log packet
+        // A log data packet is a minimum of 5 bytes. Ignore anything less.
+        if (rx_payload_len >= 5) {
+            // Port 5 = log, Channel 2 = data
+            if (rx_packet[0] == crtp_create_header(CRTP_PORT_LOG, CRTP_LOG_CHAN_LOGDATA)) {
+                // The log block ID
+                if (rx_packet[1] == CFLIE_TELEM_LOG_BLOCK_ID) {
+                    // Bytes 6 and 7 are the Vbat in mV units
+                    u16 vBat;
+                    memcpy(&vBat, &rx_packet[5], sizeof(u16));
+                    Telemetry.value[TELEM_DSM_FLOG_VOLT2] = (s32) (vBat / 10); // The log value expects tenths of volts
```

Comment:
Do you have cflie 1? I made a modification analogous to yours to add battery voltage to telemetry report, but still no luck. Maybe I don't get any telemetry because my nRF24 module is wired incorrectly, I will check this also.